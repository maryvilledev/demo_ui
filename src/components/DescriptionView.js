import React from 'react'

const DescriptionView = ({description, active}) => {
  if (active) return (
  <div>
    <h1>{description.concept_name}</h1>
    <p>{description.description}</p>
    <pre>{description.syntax}</pre>
  </div>
  )
  else return (
    <h2> Hover over code to see a description of a concept here! </h2>
  );
}

DescriptionView.propTypes = {
  description: React.PropTypes.object,
  active: React.PropTypes.bool,
}

export default DescriptionView
