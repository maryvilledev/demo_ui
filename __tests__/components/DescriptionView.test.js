import React from 'react'
import ReactDOM from 'react-dom'
import DescriptionView from '../../src/components/DescriptionView'

import testRenderer from 'react-test-renderer'

const mockDescription = {
  concept_name: "on-the-cob",
  description: "The best kind of corn.",
  syntax: "shuck+boil",
}
const mockActive = true;

describe('<DescriptionView />', () => {
  it('renders correctly', () => {
    const tree = testRenderer.create(
      <DescriptionView
        description={mockDescription}
        active={mockActive}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
