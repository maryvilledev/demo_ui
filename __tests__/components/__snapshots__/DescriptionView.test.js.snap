exports[`<DescriptionView /> renders correctly 1`] = `
<div>
  <h1>
    on-the-cob
  </h1>
  <p>
    The best kind of corn.
  </p>
  <pre>
    shuck+boil
  </pre>
</div>
`;
